packages/server/lib/plugins/child/run_plugins.js

```diff
@@ -124,7 +128,7 @@ class RunPlugins {
     .tap(() => {
       if (!this.registeredEventsByName['file:preprocessor']) {
         debug('register default preprocessor')
-        registerChildEvent('file:preprocessor', this._getDefaultPreprocessor(initialConfig))
+        registerChildEvent('file:preprocessor', this._getDefaultPreprocessor(initialConfig), true)
       }
```

Comment:
Instead of updating `registerChildEvent` which seems odd since we would need to update all of our `registerChildEvent` calls to pass `true`, can we do something like:

```suggestion
        registerChildEvent('file:preprocessor', this._getDefaultPreprocessor(initialConfig), true)
      } else {
        const handler = this.registeredEventsById[this.registeredEventsByName['file:preprocessor']].handler
        this.ipc.send('file:preprocessor:overridden', { handlerString: handler.toString() })
      }
```

Reply:
I like it. Updated: [`1a261e2` (#31171)](https://github.com/cypress-io/cypress/pull/31171/commits/1a261e2155b4545d9eca976656044d8ff8213c31)